### ISSUE TITLE
[討論議題]20200326

### ISSUE THREAD

**Issue by kcwanglucky** [sebastien911/IFNLP#6]

**Comment by kcwanglucky** [comment]
Slide 55的這句話有看沒有很懂 求開示QQ
“Initialize hidden layer biases to 0 and output (or reconstruction) biases to optimal value if weights were 0” 他說的 ”if weights were 0” 是在講什麼情形呀？

**Comment by sebastien911** [comment]
> Slide 55的這句話有看沒有很懂 求開示QQ
> “Initialize hidden layer biases to 0 and output (or reconstruction) biases to optimal value if weights were 0” 他說的 ”if weights were 0” 是在講什麼情形呀？

@kcwanglucky 
這句感覺在考英文XD
我的理解是output layer那層通常就直接進softmax了，所以在做backprop的時候如果對output layer取gradient，很難更新到最後一層的bias項，所以最後一層的bias term需要用不同方法來初始化/最佳化，不能直接給0，給0就永遠更新不到它了

例如一個簡單的網路，
z1 = xW1 + b1
a1 = tanh(z1)
z2 = a1W2 + b2
a2 = softmax(z2) = y_hat 

在算gradient時需要求d(a2) / d(b2)，如果b2一開始是0，a2中就沒有b2項了
以上理解再請高手補充~~

**Comment by sebastien911** [comment]
想知道jacobian矩陣更直觀的物理意義、計算方式之類的，有人有推薦好讀的科普文章嗎

**Comment by sunchigg** [comment]
Node Intuitions on page 38想知道有套件用這概念加速嗎·～

**Comment by GitYCC** [comment]
補充資料：
https://medium.com/@karpathy/yes-you-should-understand-backprop-e2f06eab496b
https://medium.com/@chih.sheng.huang821/深度學習-weight-initialization和batch-normalization-f264c4be37f5

**Comment by STCYC** [comment]
Nice videos to illustrate Backpropagation
https://www.youtube.com/watch?v=Ilg3gGewQ5U
https://www.youtube.com/watch?v=tIeHLnjs5U8
